### PR TITLE
make positions for channels and roles nicer

### PR DESCRIPTION
**Pull request by GusCaplan**

### CODE REVIEW

**Comment by iCrawl** [comment]
This needs a rebase @GusCaplan

File: src/structures/Guild.js [hunk 1]
```diff
@@ -901,6 +877,67 @@ class Guild {
...
+
+    let updatedRoles = this._sortedRoles().array();
+
+    Util.moveElementInArray(updatedRoles, role, position, relative);
+
+    updatedRoles = updatedRoles.map((r, i) => ({ id: r.id, position: i }));
+    return this.client.rest.methods.setRolePositions(this.id, updatedRoles);
+  }
+
+  setChannelPosition(channel, position, relative = false) {
```
Comment (PgBiel) [inline src/structures/Guild.js]: I think you forgot documentation here

File: src/structures/Guild.js [hunk 2]
```diff
@@ -901,6 +877,74 @@ class Guild {
...
+
+    let updatedChannels = this._sortedChannels(channel.type).array();
+
+    Util.moveElementInArray(updatedChannels, channel, position, relative);
+
+    updatedChannels = updatedChannels.map((r, i) => ({ id: r.id, position: i }));
+    return this.client.rest.methods.setChannelPositions(this.id, updatedChannels);
+  }
+
+  _sortedRoles() {
```
Comment (Gawdl3y) [inline src/structures/Guild.js]: Getter? Documentation?

File: src/structures/Guild.js [hunk 3]
```diff
@@ -901,6 +877,74 @@ class Guild {
...
+    return this._sortPositionWithID(this.roles);
+  }
+
+  _sortedChannels(type) {
```
Comment (Gawdl3y) [inline src/structures/Guild.js]: Getter? Documentation?

File: src/structures/Guild.js [hunk 4]
```diff
@@ -901,6 +877,74 @@ class Guild {
...
+    return this._sortPositionWithID(this.channels.filter(c => {
+      if (type === 'voice' && c.type === 'voice') return true;
+      else if (type !== 'voice' && c.type !== 'voice') return true;
+      else return type === c.type;
+    }));
+  }
+
+  _sortPositionWithID(collection) {
```
Comment (Gawdl3y) [inline src/structures/Guild.js]: Documentation?

**Pull request merged**